tasks/test_results_finisher.py

```diff
@@ -296,121 +283,43 @@ def process_impl_within_lock(
-                    db_session, repoid, notifier, payload, checkpoints=checkpoints
-                )
-
-            metrics.incr(
-                "test_results.finisher.flaky_test_detection",
-                tags={"status": success, "reason": reason},
-            )
-
         return {
             "notify_attempted": True,
             "notify_succeeded": success,
             QUEUE_NOTIFY_KEY: False,
         }
 
-    def run_flaky_test_detection(
+    def get_flaky_tests(
         self,
-        db_session,
-        repoid,
-        notifier: TestResultsNotifier,
-        payload: TestResultsNotificationPayload,
-        checkpoints=None,
+        db_session: Session,
+        commit_yaml: UserYaml,
+        repoid: int,
+        commit: Commit,
+        failures: list[TestResultsNotificationFailure],
     ):
-        ignore_predefined = read_yaml_field(
-            "test_analytics", "ignore_predefined", _else=False
-        )
-
-        user_normalization_regex = read_yaml_field(
-            "test_analytics", "normalization_regex", _else=dict()
-        )
-
-        failure_normalizer = FailureNormalizer(
-            user_normalization_regex, ignore_predefined
-        )
-
-        default_branch_failure_detector = DefaultBranchFailureDetector(
-            db_session, repoid, "main"
-        )
-        unrelated_matches_detector = UnrelatedMatchesDetector(failure_normalizer)
-        diff_outcome_detector = DiffOutcomeDetector()
-
-        flake_detection_engine = FlakeDetectionEngine(
-            db_session,
-            repoid,
-            [
-                default_branch_failure_detector,
-                unrelated_matches_detector,
-                diff_outcome_detector,
-            ],
-        )
-
-        log.info(
-            "Starting flake detection",
-            extra=dict(
-                repoid=repoid,
-                parent_task=self.request.parent_id,
-            ),
-        )
-        current_state_of_repo_flakes = flake_detection_engine.detect_flakes()
-
-        for test_id, symptoms in current_state_of_repo_flakes.items():
-            log.info(
-                "Discovered flaky test",
-                extra=dict(
-                    repoid=repoid,
-                    parent_task=self.request.parent_id,
-                    test_id=test_id,
-                    symptoms=list(symptoms),
-                ),
-            )
-            payload.flaky_tests[test_id] = TestResultsNotificationFlake(
-                list(symptoms),
-                True,
+        if FLAKY_TEST_DETECTION.check_value(
+            identifier=repoid, default=False
+        ) and read_yaml_field(
+            commit_yaml, ("test_analytics", "flake_detection"), False
+        ):
+            flaky_tests = set()
+            repo_flakes = (
+                db_session.query(Flake)
+                .filter(  # type:ignore
+                    Flake.repoid == repoid,
+                    Flake.end_date.is_(None),
```

Comment:
I see that the index is added on a compound of ["repository", "test", "reduced_error", and "date"]. This means that for this query, it's _only_ going to use the index for "repository" (because [Postgres processes index from left to right](https://www.postgresql.org/docs/current/indexes-multicolumn.html)). 

I _believe_ we can also add an index for null fields specifically to make it more efficient, but I don't have as much experience with this. :p If we want to make this query efficient, we might want to explore doing a compound index with (repository, enddate = null) or something.

Reply:
would adding on a `Flake.testid.in_(list_of_test_ids)` make this more efficient then? I plan to to eventually add the reduced_error_id to the query as well, so in the end we will be making full use of this index.

Reply:
> I plan to to eventually add the reduced_error_id to the query as well, so in the end we will be making full use of this index.

Ah OK! This is fine then. 👍